clients/spark/core/src/main/scala/io/treeverse/clients/GarbageCollector.scala

```diff
@@ -547,28 +549,42 @@ object GarbageCollector {
   private def getPreviousRunID(
       incrementalRun: Boolean,
       storageNS: String,
-      configMapper: ConfigMapper
+      configMapper: ConfigMapper,
+      iterations: Int
   ): String = {
     if (!incrementalRun) {
       return ""
     }
+    if (iterations < 1) {
+      throw RunIDException("Run ID iteration number cannot be smaller than 1")
+    }
     var previousRunID = ""
-    val runIDsPath = new Path(s"${storageNS.stripSuffix("/")}/_lakefs/retention/gc/run_ids/")
+    val runIDsPath = new Path(
+      String.format(RUN_ID_MARKERS_LOCATION_FORMAT, storageNS.stripSuffix("/"), "")
+    )
     val runIDsFS = runIDsPath.getFileSystem(configMapper.configuration)
     val runIDs = runIDsFS.listFiles(runIDsPath, false)
-    if (runIDs.hasNext) {
-      previousRunID = runIDs.next().getPath.getName
-      println(s"----------------------- Using previous RUN ID: $previousRunID")
-    } else {
+    if (!runIDs.hasNext) {
       runIDsFS.close()
       throw RunIDException("No previous run ID")
     }
+    var runIDObject: LocatedFileStatus = null
+    for (_ <- 0 until iterations) {
+      if (!runIDs.hasNext) {
+        throw RunIDException("Required Run ID iteration doesn't exist")
+      }
+      runIDObject = runIDs.next()
+    }
```

Comment:
If we don't use a proper (Scala) Iterator, can we at least move this out into a separate function?  "Get the nth element of a RemoteIterator" sounds like a description of a testable function.